### DISCUSSION TITLE
Any webpack example?

### DISCUSSION THREAD

**Discussion by mokone91** [vueuse/vue-demi#82] [Q&A]
Hi! 
im working on components lib and truing to make it vue2/3 compatible using this solution.
But i faced with issue with vue3 installation.. in vue2 env everything working fine!
 Here is my build webpack config: 
```
import merge from 'webpack-merge';
import { Configuration } from 'webpack';
import { CleanWebpackPlugin } from 'clean-webpack-plugin';
import common from './webpack.base.config';
const { watch } = require('minimist')(process.argv.slice(2));

const path = require('path');


export default merge<Configuration>(common, {
  entry: {
    main: './src/index.ts',
  },
  output: {
    filename: 'index.js',
    path: path.resolve(__dirname, 'dist'),
    library: {
      name: 'TestLib',
      type: 'umd',
    },
  },
  devtool: 'source-map',
  mode: watch ? 'development' : 'production',
  bail: watch ? false : true,
  plugins: [
    new CleanWebpackPlugin(),
  ],
  externals: {
    vue: 'vue',
    'vue-demi': 'vue-demi',
    'lodash': 'lodash',
  },
});
```
on client side im using default vue-cli-service serve^ without additional settings
but in vue3 environment i got:
![image](https://user-images.githubusercontent.com/2384641/127203132-d639ba05-ea70-4b13-9424-2141344192d8.png)

seems problem somewhere near webpack build config, maybe somebody can share some webpack config examples? 
Thanks!

**Comment by mokone91** [comment]
upd:
Originally components lib done in vue2, if i change my dev deps like: 
```diff
  -"@vue/composition-api": "^1.0.4",
  +"@vue/compiler-sfc": "^3.1.5",
  -"vue": "^2.6.4",
  +"vue": "^3.1.5",
  -"vue-loader": "^15.9.6",
  +"vue-loader": "^16.3.3",
  -"vue-template-compiler": '2.6.14',
```

Its going to work in vue3 env, but broken in vue2 with: 
![image](https://user-images.githubusercontent.com/2384641/127293667-91e8af89-7395-489d-8225-838b77e53fd2.png)

seems like vue bundled in even if i put it to externals 

upd:
as i can see in bundle analyzer - there is vue-style-loader bundled only
![image](https://user-images.githubusercontent.com/2384641/127293936-c52e3d6d-8659-49e1-b916-21d0498e8ccb.png)


upd:
After removing all 'scoped' keywords start getting next error: 
![image](https://user-images.githubusercontent.com/2384641/127301181-c48a1803-7a3f-4336-9e22-78db9fa51678.png)

**Comment by antfu** [comment]
I don't have much experience for configuring Webpack, so I might not be able to help. Since this lib works fine in Vue CLI, I think it does not have compatibility issues with Webpack. Converted to discussion if any one want to help.

**Reply by mokone91** [reply]
Hi, have found prev conversation https://github.com/vueuse/vue-demi/issues/65
seems problem can't be solved by simple config fix, as im truing to make it work with regular template section

**Reply by mokone91** [reply]
just wondering now is its possible to replace vue render function with h-demi somewhere in the middle of webpack build

**Reply by mokone91** [reply]
so, decided to make 2 builds with only difference in vue-loader(15/16)
